dfget/core/core.go

```diff
@@ -189,8 +190,36 @@ func downloadFile(cfg *config.Config, supernodeAPI api.SupernodeAPI,
 	reportFinishedTask(cfg, getter)
 
 	os.Remove(cfg.RV.TempTarget)
-	logrus.Infof("download %s cost:%.3fs length:%d reason:%d",
-		success, time.Since(cfg.StartTime).Seconds(), cfg.RV.FileLength, cfg.BackSourceReason)
+	downloadTime := time.Since(cfg.StartTime).Seconds()
+	if cfg.UploadMetrics {
+		req := &types.TaskMetricsRequest{
+			BacksourceReason: strconv.Itoa(cfg.BackSourceReason),
+			IP:               cfg.RV.LocalIP,
+			CID:              cfg.RV.Cid,
+			CallSystem:       cfg.CallSystem,
+			Duration:         downloadTime,
+			FileLength:       cfg.RV.FileLength,
+			Port:             int32(cfg.RV.PeerPort),
+			Success:          success,
+			TaskID:           result.TaskID,
+		}
+		for _, node := range cfg.Node {
+			resp, err := supernodeAPI.ReportMetrics(node, req)
```

Comment:
I am not sure about this part. In the offline discuss with @lowzj and @Starnop , we agreed that it is OK that dfget upload to one supernode. I am wondering here if we need some shuffle or hash mechanism to distribute api requests? Otherwise dfget will always upload to one supernode